File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        nums = [0] * n
        for i in range(n):
            if i == n - 1:
                nums[i] = -1
            else:
                nums[i] = max(arr[i+1:n])
        
        return nums

File: Data Structures & Algorithms/house-robber-ii/submission-1.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        match nums.len() {
            1 => nums[0],
            n => {
                let skip_first = Self::helper(&nums[1..]);
                let skip_last = Self::helper(&nums[..n-1]);
                skip_first.max(skip_last)
            },
        }
    }
    fn helper(nums: &[i32]) -> i32 {
        let (mut rob1, mut rob2) = (0i32, 0i32);
        for n in nums {
            let temp = (n + rob1).max(rob2);
            rob1 = rob2;
            rob2 = temp
        }
        rob2
    }
}
